src/wifipaf/WiFiPAFLayer.h

```diff
@@ -0,0 +1,204 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "WiFiPAFConfig.h"
+#include "WiFiPAFEndPoint.h"
+#include "WiFiPAFLayerDelegate.h"
+#include "WiFiPAFRole.h"
+#include <lib/core/CHIPError.h>
+#include <lib/support/DLLUtil.h>
+#include <system/SystemLayer.h>
+#include <system/SystemPacketBuffer.h>
+
+namespace chip {
+namespace WiFiPAF {
+
+/**
+ *  @def NUM_PAFTP_SUPPORTED_PROTOCOL_VERSIONS
+ *
+ *  Number of unsigned 4-bit representations of supported transport protocol
+ *  versions encapsulated in a BleTransportCapabilitiesRequest. Defined by CHIP
+ *  over PAFTP protocol specification.
+ */
+#define NUM_PAFTP_SUPPORTED_PROTOCOL_VERSIONS 8
+
+// Version(s) of the CHIP PAF Transport Protocol that this stack supports.
+#define CHIP_PAF_TRANSPORT_PROTOCOL_MIN_SUPPORTED_VERSION kWiFiPAFTransportProtocolVersion_V1
+#define CHIP_PAF_TRANSPORT_PROTOCOL_MAX_SUPPORTED_VERSION kWiFiPAFTransportProtocolVersion_V1
+
+/// Enum defining versions of CHIP over PAF transport protocol.
+typedef enum
+{
+    kWiFiPAFTransportProtocolVersion_None = 0,
+    kWiFiPAFTransportProtocolVersion_V1   = 4 // PAFTP as defined by CHIP v1.5
+} WiFiPAFTransportProtocolVersion;
+
+inline constexpr size_t kCapabilitiesRequestMagicnumLength          = 2;
+inline constexpr size_t kCapabilitiesRequestL2capMtuLength          = 2;
+inline constexpr size_t kCapabilitiesRequestSupportedVersionsLength = 4;
```

Comment:
Why isn't this computed from NUM_PAFTP_SUPPORTED_PROTOCOL_VERSIONS?  Or at least static_asserted to match it?

Reply:
I will change to compute it from NUM_PAFTP_SUPPORTED_PROTOCOL_VERSIONS